tracing/tracing-test.sh

```diff
@@ -23,17 +32,48 @@ jaeger_server=${jaeger_server:-localhost}
 jaeger_ui_port=${jaeger_ui_port:-16686}
 jaeger_docker_container_name="jaeger"
 
+logdir=""
+
 # Cleanup will remove Jaeger container and
 # disable tracing.
-cleanup(){
-	stop_jaeger 2>/dev/null || true
-	"${SCRIPT_PATH}/../.ci/configure_tracing_for_kata.sh" disable
-}
+cleanup()
+{
+	local fp="info"
+	local result="failed"
+	local dest="$logdir"
+
+	if [ "$success" = true ]; then
+		result="passed"
+
+		[ -z "$DEBUG_KEEP_JAEGER" ] && stop_jaeger 2>/dev/null || true
 
-trap cleanup EXIT
+		# The tests worked so remove the logs
+		if [ -n "$DEBUG" ]; then
+			eval "$fp" "test $result - logs left in '$dest'"
+		else
+			"${SCRIPT_PATH}/../.ci/configure_tracing_for_kata.sh" disable
+
+			[ -d "$logdir" ] && rm -rf "$logdir" || true
+		fi
+
+		return 0
+	fi
+
+	if [ -n "${CI:-}" ]; then
+		# Running under the CI, so copy the logs to allow them
+		# to be added as test artifacts.
+		sudo mkdir -p "$TRACE_LOG_DIR"
+		sudo cp -a "$logdir"/* "$TRACE_LOG_DIR"
+
+		dest="$TRACE_LOG_DIR"
+		fp="die"
+	fi
+
+	eval "$fp" "test $result - logs left in '$dest'"
```

Comment:
If `success==false` and not in `CI` mode then it won't `die`.

Reply:
Fixed.